src/world/action/jump.c

```diff
@@ -1,15 +1,250 @@
 #include "common.h"
 
-INCLUDE_ASM(void, "world/action/jump", func_802B6000_E24040, void);
+extern void phys_adjust_cam_on_landing();
+extern void sfx_play_sound_at_player(s32 soundID, s32 arg0);
+extern void suggest_player_anim_clearUnkFlag(s32 arg0);
+extern void phys_init_integrator_for_current_state();
+extern s32 check_input_jump();
+extern s32 player_raycast_below_cam_relative(PlayerStatus*, f32*, f32*, f32*, f32*, f32*, f32*, f32*, f32*);
 
-INCLUDE_ASM(void, "world/action/jump", func_802B60B4_E240F4, void);
+extern f32 D_8010C960;
+extern f32 D_8010C97C;
 
-INCLUDE_ASM(void, "world/action/jump", func_802B6198_E241D8, void);
+void func_802B6000_E24040() {
+    s32 temp_v1;
+    s32 phi_a0;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    CollisionStatus* collisionStatus;
 
-INCLUDE_ASM(void, "world/action/jump", func_802B6294_E242D4, void);
+    playerStatus->fallState = 0;
+    playerStatus->decorationList = 0;
+    playerStatus->unk_C2 = 0;
+    playerStatus->flags &= 0x7FFFFFF7 ;
+    playerStatus->flags |= 2;
+    playerStatus->unk_3C = playerStatus->position.x;
+    playerStatus->unk_40 = playerStatus->position.z;
+    playerStatus->unk_4C = playerStatus->position.y;
 
-INCLUDE_ASM(void, "world/action/jump", func_802B6348_E24388, void);
+    phys_init_integrator_for_current_state();
+    collisionStatus = &gCollisionStatus;
 
-INCLUDE_ASM(void, "world/action/jump", func_802B647C_E244BC, void);
+    if ((playerStatus->animFlags & PLAYER_ANIM_FLAG_8BIT_MARIO)) {
+        temp_v1 = 0x90005;
+    }
+    else {
+        phi_a0 = (playerStatus->animFlags & PLAYER_ANIM_FLAG_3);
```

Comment:
This can be flag 2 bitwise ored with flag 1 as mentioned above. Also, I don't think you need this temp and you can just put the expression into the if condition. You may have to invert the condition for it to match